Update react-native.d.ts

chrisfisher opened the pull request:
Added additional props to NavigationHeaderProps.

bgrieder commented:
@chrisfisher Hi. Thank you for this.

`NavigationHeaderProps` does need better typings but, since we are at it, it seems there is a bit more than what you just added (such as `renderRightComponent`), see: https://github.com/facebook/react-native/blob/master/Libraries/CustomComponents/NavigationExperimental/NavigationHeader.js#L68

Do you mind adding them ?

Also is there a chance that `style` is typed to something a bit better than `any` ?

chrisfisher commented:
@bgrieder Thanks for the quick response!

I've added the additional props - sorry, probably should have checked a bit more thoroughly the first time.

I also updated the type for NavigationHeaderStatic.Title from:

`Title: JSX.Element`

to:

`Title: () => JSX.ElementClass`

Otherwise I get this error when trying to compile typescript:

'JSX element type 'NavigationHeaderTitle' does not have any construct or call signatures.'

But I'm pretty new to TypeScript & NavigationExperimental so might be doing something wrong.

This is my implementation (the relevant part of it):

```
import { NavigationExperimental } from 'react-native';

const { Header: NavigationHeader } = NavigationExperimental;
const { Title: NavigationHeaderTitle } = NavigationHeader;

class NavHeader extends Component<Props, {}> {
  render() {
    return (
      <NavigationHeader
        renderTitleComponent={this._renderTitleComponent.bind(this)}
        ...
      />
    );
  }

  _renderTitleComponent() {
    return (
      <NavigationHeaderTitle textStyle={styles.headerTitle}>
        {this.props.scene.route.title}
      </NavigationHeaderTitle>
    );
  }
}

```

bgrieder commented:
👍
